docs/writing-tests/testharness.md

```diff
@@ -198,15 +182,83 @@ Use `// META: script=link/to/resource.js` at the beginning of the resource. For
 
 can be used to include both the global and a local `utils.js` in a test.
 
-### Specifying a timeout of long in auto-generated boilerplate tests
+### Specifying a timeout of long
 
 Use `// META: timeout=long` at the beginning of the resource.
 
-### Specifying test [variants](#variants) in auto-generated boilerplate tests
+### Specifying test [variants](#variants)
 
 Use `// META: variant=url-suffix` at the beginning of the resource. For example,
 
 ```
 // META: variant=
 // META: variant=?wss
 ```
+
+## Variants
+
+A test file can have multiple variants by including `meta` elements,
+for example:
+
+```html
+<meta name="variant" content="">
+<meta name="variant" content="?wss">
+```
+
+The test can then do different things based on the URL.
```

Comment:
This explanation glosses over an important part of the feature: it doesn't connect the elements to the URL. For instance, "Test runners will execute the test for each 'variant' specified, appending the corresponding `content` value to the URL as they do so." I recognize this part hasn't changed, but would you mind adding something to that effect while we're here?